datastream-jdbc-connector/src/main/java/com/linkedin/datastream/connectors/jdbc/translator/JdbcCommon.java

```diff
@@ -641,6 +632,15 @@ public static Schema createSchema(final ResultSet rs, AvroConversionOptions opti
                                     : u.stringType());
                     break;
 
+                case TIMESTAMP:
+                case TIMESTAMP_WITH_TIMEZONE:
+                case microsoft.sql.Types.SMALLDATETIME: // -150 represents TSQL smalldatetime type
+                case microsoft.sql.Types.DATETIME: // -151 represents TSQL datetime type
+                case microsoft.sql.Types.DATETIMEOFFSET: // TSQL DATETIMEOFFSET
+                    final Schema timestampMilliType = LogicalTypes.timestampMillis().addToSchema(SchemaBuilder.builder().longType());
```

Comment:
You will have to add an additional attribute/property to the schema that will let the consumer identify the source datatype.

https://avro.apache.org/docs/1.7.2/api/java/org/apache/avro/Schema.Field.html#addProp(java.lang.String,%20java.lang.String)

The challenge is if we convert all these types to logical type timestampmillis without adding additional info about the source field's datatype, then the consumer may not be able to tell if the time is adjusted UTC accurately or not.

Introduce a new property something like "source.sql.datetype" and set it to "datetime", "datetimeoffset"...